tests/validate-urls.rb

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'English'
+require 'json'
+@status = 0
+
+# Fetch created/modified files in entries/**
+diff = `git diff --name-only --diff-filter=AM origin/master...HEAD entries/`.split("\n")
+
+def curl(url)
+  puts `curl --fail -sSI #{url}`
+  # rubocop:disable Style/GuardClause
+  unless $CHILD_STATUS.success?
+    # Break build if above cURL exited with non-zero value
+    puts "::error file=#{@path}:: Unable to reach #{url}"
+    @status = 1
+  end
+  # rubocop:enable Style/GuardClause
+end
+
+diff.each do |path|
+  # Make path global for curl()
+  @path = path
+  entry = JSON.parse(File.read(@path)).values[0]
+
+  # Process the url,domain & additional-domains
+  curl((entry.key?('url') ? entry['url'] : "https://#{entry['domain']}/"))
+  entry['additional-domains'].each { |domain| curl("https://#{domain}/") }
```

Comment:
```suggestion
  entry['additional-domains'].each { |domain| curl("https://#{domain}/") } if entry.key? 'additional-domains'
```
Otherwise I get ```undefined method `each' for nil:NilClass``` for entries without `additional-domains`. Maybe that is related to the `nil` problem you had before?

Reply:
What a weird language Ruby is. I would have bet on it just skipping the iteration if the the array is nil.

Apparently `&` does the same thing as `if entry.key? 'additional-domains'`.